pip/qsharp/_qsharp.py

```diff
@@ -6,28 +6,40 @@
 _interpreter = None
 
 
-def init(target_profile: TargetProfile = TargetProfile.Full) -> None:
+def init(
+    *, target_profile: TargetProfile = TargetProfile.Full, project_root=None
+) -> None:
     """
     Initializes the Q# interpreter.
 
     :param target_profile: Setting the target profile allows the Q#
         interpreter to generate programs that are compatible
         with a specific target. See :py:class: `qsharp.TargetProfile`.
+
+    :param project_root: The root directory of the Q# project. It must
+        contain a qsharp.json project manifest.
     """
     global _interpreter
-    _interpreter = Interpreter(target_profile)
 
+    manifest_descriptor = None
+    if project_root is not None:
+        import os
 
-def get_interpreter() -> Interpreter:
-    """
-    Returns the Q# interpreter.
+        qsharp_json = os.path.join(project_root, "qsharp.json")
+        if not os.path.exists(qsharp_json):
+            raise ValueError("qsharp.json not found at project root")
 
-    :returns: The Q# interpreter.
-    """
-    global _interpreter
-    if _interpreter is None:
-        init()
-    return _interpreter
+        import json
+
+        manifest_descriptor = {}
+        manifest_descriptor["manifest_dir"] = project_root
+        manifest_descriptor["manifest"] = json.loads(
+            open(qsharp_json, mode="r", encoding="utf-8").read()
+        )
```

Comment:
1. Wihout a corresponding close() call the file is left open. The `with open(...) as f:` construct handles this closing the file automatically.
2. Do we want to provide a custom error message to the user if the project file can't be read?
3. There is a util function at the bottom of the file that does the same read logic, do we want to consolidate?